From patchwork Wed Aug 29 05:54:47 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: Wanpeng Li <kernellwp@gmail.com>
X-Patchwork-Id: 10579547
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 2211A14BD
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 29 Aug 2018 05:55:25 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 0A45329203
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 29 Aug 2018 05:55:25 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id F2AD92922D; Wed, 29 Aug 2018 05:55:24 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A538A29203
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 29 Aug 2018 05:55:24 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727098AbeH2Jue (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 29 Aug 2018 05:50:34 -0400
Received: from mail-pg1-f178.google.com ([209.85.215.178]:41742 "EHLO
        mail-pg1-f178.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726656AbeH2Jue (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 29 Aug 2018 05:50:34 -0400
Received: by mail-pg1-f178.google.com with SMTP id s15-v6so1840989pgv.8
        for <kvm@vger.kernel.org>; Tue, 28 Aug 2018 22:55:22 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=HW4NoCpfirTNvO1ri611hPMmzKpzDsFrtRJrMrWussI=;
        b=WIrVaWpUkC+qNZwnozzMpyG4rlzXDs1Wmj7Duwa9ykGSEVmlQ/eVFu26jodw2i4k/G
         /JnQg+ys35t0xI6YBq5w8mI8L35yVWlU8miqn48C5rsmjie97uXvmyMVlm+lTB3isCXz
         HemTi7d5J5XKW7UXBQAWFFFBjmbjosDyjYFNiWWrIEFtHAdeFaeTUdU6JKZazsF/qQLI
         aWtetNcrOxWLRuZSguu0xeOBeluKZuuMJ2ZntODF1Gw9eqOOEk5nBBfSVMsqAmy1IPvu
         xB9VOO01ZJaBuWqRuqinqB/WhKZkgShg7Byc0VN+bnJF5fEApuwD462QtjC6PhiVjsrE
         LzVQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=HW4NoCpfirTNvO1ri611hPMmzKpzDsFrtRJrMrWussI=;
        b=uVVifwjgMwtGq/DMU0MCLAgk9kpaecQ0NCTTOkXrIilGpR6C6gTKJBCn+G2Iu+rdVU
         wu4Kl1R/+S6Ssj6dxQnOjXB2KA+TKXtfhXL1pDbdTim5DXXmDsNxZrSq/6q7jT2WScHp
         OaxJWiMpWMtBelCiLdUlqjEn408F0PCHxZ2LEVyaZqdq79DKkuO/OAI5oDRdGNynYI9V
         UKL9h03j5HVLnm+C4cTishyc5Xf3/luQ25vrLieVXhY5jFNqet0x2PAB1u07SD3ROfZg
         5Nhk+7WpzndLUjN/VccsSpoiQAMx84WQYhub/xavT+icKaxd13vloEJzrk4q8AbpyG0a
         ReYA==
X-Gm-Message-State: APzg51CzK7599oBBjDN/YXW4YtRvQvLpPzVtpAiwD7M2e7fvCuPUdJ85
        fVfiDRFZYVhkRzVro0hTp5e/IO2k
X-Google-Smtp-Source: 
 ANB0VdaZr3yNBBY5mK998zZVNwoD22Wtzvv64aUubDREjRsUd6H1yMjfDxf+OtzEm7XexNBEHGws6w==
X-Received: by 2002:a62:490e:: with SMTP id
 w14-v6mr4421915pfa.213.1535522122463;
        Tue, 28 Aug 2018 22:55:22 -0700 (PDT)
Received: from localhost.localdomain ([203.205.141.123])
        by smtp.googlemail.com with ESMTPSA id
 v26-v6sm3957369pfe.57.2018.08.28.22.55.20
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-SHA bits=128/128);
        Tue, 28 Aug 2018 22:55:22 -0700 (PDT)
From: Wanpeng Li <kernellwp@gmail.com>
X-Google-Original-From: Wanpeng Li <wanpengli@tencent.com>
To: kvm@vger.kernel.org
Cc: Paolo Bonzini <pbonzini@redhat.com>,
 =?utf-8?b?UmFkaW0gS3LEjW3DocWZ?= <rkrcmar@redhat.com>,
 Liran Alon <liran.alon@oracle.com>
Subject: [PATCH kvm-unit-tests] x86: apic: pv ipis out-of-bounds access
 testing
Date: Wed, 29 Aug 2018 13:54:47 +0800
Message-Id: <1535522087-5704-1-git-send-email-wanpengli@tencent.com>
X-Mailer: git-send-email 2.7.4
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Wanpeng Li <wanpengli@tencent.com>

Adds pv ipis out-of-bounds acess testing. 

Cc: Paolo Bonzini <pbonzini@redhat.com>
Cc: Radim Krčmář <rkrcmar@redhat.com>
Cc: Liran Alon <liran.alon@oracle.com>
Signed-off-by: Wanpeng Li <wanpengli@tencent.com>
Reviewed-By: Liran Alon <liran.alon@oracle.com>
---
 x86/apic.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/x86/apic.c b/x86/apic.c
index c976e8d..460d7a5 100644
--- a/x86/apic.c
+++ b/x86/apic.c
@@ -566,6 +566,15 @@ static void test_apic_change_mode(void)
 	report("TMCCT should stay at zero", !apic_read(APIC_TMCCT));
 }
 
+void test_pv_ipi()
+{
+    unsigned int nr = 10, ret;
+    unsigned long p1 = 0xFFFFFFFF, p2 = 0, p3 = 0xFFFFFFFF, p4 = 0x0;
+
+    asm volatile("vmcall" : "=a"(ret) :"a"(nr), "b"(p1), "c"(p2), "d"(p3), "S"(p4));
+    report("PV IPIs testing", !ret);
+}
+
 int main(void)
 {
     setup_vm();
@@ -581,6 +590,7 @@ int main(void)
 
     test_self_ipi();
     test_physical_broadcast();
+    test_pv_ipi();
 
     test_sti_nmi();
     test_multiple_nmi();
